File: 0648-replace-words/0648-replace-words.py
class Solution:
    def replaceWords(self, dictionary: List[str], sentence: str) -> str:
        words = sentence.split()
        for i in range(len(words)):
            for root in dictionary:
                if not words[i].startswith(root):
                    continue
                words[i] = root
        return " ".join(words)

File: 0648-replace-words/0648-replace-words.js
/**
 * @param {string[]} dictionary
 * @param {string} sentence
 * @return {string}
 */
var replaceWords = function(dictionary, sentence) {
  let a = sentence.split(" ");
  let b = dictionary;

  for (let i = 0; i < a.length; i++) {
    for (let j = 0; j < b.length; j++) {
      if (a[i].startsWith(b[j])) {
        a[i] = b[j];
      }
    }
  }

  return a.join(" ");
};